shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/rule/DataNodesTest.java

```diff
@@ -79,18 +77,19 @@ public void assertGetDataNodeGroups() {
     }
     
     private DataNodes getRoutedRuleDataNodes() {
-        TestTableRule tableRule1 = new TestTableRule(dataSourceNames1, logicTableName1);
-        TestTableRule tableRule2 = new TestTableRule(dataSourceNames2, logicTableName2);
-        List<TestTableRule> tableRules = Arrays.asList(tableRule1, tableRule2);
-        ShardingSphereRule rule1 = new TestShardingRule(tableRules);
+        Map<String, Collection<DataNode>> nodeMap = new HashMap<>();
+        nodeMap.put(logicTableName1, getExpectedDataNodes(dataSourceNames1, logicTableName1));
+        nodeMap.put(logicTableName2, getExpectedDataNodes(dataSourceNames2, logicTableName2));
+        DataNodeContainedRule rule1 = mock(DataNodeContainedRule.class);
+        when(rule1.getAllDataNodes()).thenReturn(nodeMap);
         Map<String, Collection<String>> dataSourceMapper = Collections.singletonMap(logicDataSourceName, replicaDataSourceNames);
         DataSourceContainedRule rule2 = mock(DataSourceContainedRule.class);
         when(rule2.getDataSourceMapper()).thenReturn(dataSourceMapper);
         return new DataNodes(Arrays.asList(rule1, rule2));
     }
     
     private DataNodes getNonRoutedRuleDataNodes() {
-        return new DataNodes(Collections.singleton(new TestShardingSphereRule()));
```

Comment:
Hi @wenweibin 
Do you think `TestShardingSphereRule` is no longer needed? If so, please delete them. Thanks.

Reply:
Yes.I will delete them.

Reply:
You will remove them in the next PR, won't you?

Reply:
OK.I will finish it in next pr.